nengo_spa/modules/decode.py

```diff
@@ -0,0 +1,39 @@
+import nengo
+from nengo.config import Default
+from nengo.params import IntParam
+
+from nengo_spa.modules.transcode import TranscodeFunctionParam
+from nengo_spa.network import Network
+from nengo_spa.vocab import VocabularyOrDimParam
+
+
+class Decode(Network):
+    function = TranscodeFunctionParam(
+        'function', sp_output=False, optional=False, default=None,
+        readonly=True)
+    vocab = VocabularyOrDimParam(
+        'vocab', optional=False, default=None, readonly=True)
+    size_out = IntParam(
+        'size_out', low=0, optional=True, default=None, readonly=True)
+
+    def __init__(
+            self, function=Default, vocab=Default, size_out=Default, **kwargs):
+        super(Decode, self).__init__(**kwargs)
+
+        self.vocab = vocab  # needs to be set before function
```

Comment:
The validation of the `TranscodeFunctionParam` will check the vocabulary dimensionality.